### ISSUE TITLE
`JsonWebKey2020` is not defined for JWS / JWT

### ISSUE THREAD

**Issue by letmaik** [decentralized-identity/did-jose-extensions#9]
`JsonWebKey2020` is defined in https://w3c-ccg.github.io/lds-jws2020, quoting:
> This specification defines a cryptographic suite for the purpose of creating, verifying proofs for JSON Web Signatures in conformance with the Linked Data Proofs [LD-PROOFS] specification. [...] The JSON Web Signature signature suite 2020 MUST be used in conjunction with the signing and verification algorithms in the Linked Data Proofs [LD-PROOFS] specification.

In summary, `JsonWebKey2020` can only be used for validating linked data signatures, not JWS / JWT.

What am I missing? The README seems to suggest otherwise.

**Comment by letmaik** [comment]
@OR13 @oed @mirceanis Any comment? I see the same issue for other suites like `EcdsaSecp256k1VerificationKey2019` which are used for JWT validation in the [did-jwt](https://github.com/decentralized-identity/did-jwt/blob/0d2918a124db9d86c259196822c1520bed0d1563/src/JWT.ts#L95) library.

**Comment by OR13** [comment]
> In summary, JsonWebKey2020 can only be used for validating linked data signatures, not JWS / JWT.

that suite IS validating (detached) JWS.... I think the linked data signature suite spec language could be better.

What the suite spec is trying to say (based on CCG conventions) is that:

JsonWebSignature2020 -> is validated by -> JsonWebKey2020.

not that `JsonWebKey2020` can ONLY be used to validate `JsonWebSignature2020`....

In practice we use `JsonWebKey2020` keys (really publicKeyJwk and privateKeyJwk) to sign and verify JWS  / JWT. 

JOSE does not care about fields outside of `publicKeyJwk` or `privateKeyJwk`... like `id`, `controller`, `type`.

By convention some folks like to use `kid` == `verificationMethod.id`... because it provides a way to discover the public key for the JWS.

**Comment by letmaik** [comment]
@OR13 I see, thanks for clarifying. I guess I expected `JsonWebKey2020` to live in a separate spec independent of a suite spec. I think calling it "verification method" is quite confusing, given that it is just verification material (or more generally parameters) used together with an externally defined verification process/algorithm. On the other hand, https://w3c-ccg.github.io/verifiable-conditions/ seems to be a mix of process and material.
I guess a JsonWebSigning20xx suite spec is missing, assuming it even makes sense to write such a thing, given that details are often application-specific.

**Comment by OR13** [comment]
@letmaik yes, sadly, the CCG thinks of things a bit differently than JOSE / IANA.

You can view all JOSE assignments here: https://www.iana.org/assignments/jose/jose.xhtml

For DIDs, we have https://w3c.github.io/did-spec-registries/#jsonwebkey2020

Since DID Core works for both `application/did+json` and `application/did+ld+json` it's important that both representations handle JWKs the same way.... for `application/did+json`, you can just ignore the `@context` if present.

for `application/did+ld+json` you will want to make sure your did document contains a context which defines that key type....

```
"@context": [
    "https://www.w3.org/ns/did/v1",
    "https://w3id.org/security/suites/jws-2020/v1"
  ],
```
